src/main/java/org/spongepowered/common/data/holder/SimpleNbtDataHolder.java

```diff
@@ -22,27 +22,37 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.common.data.holder.nbt;
+package org.spongepowered.common.data.holder;
 
 import net.minecraft.nbt.CompoundNBT;
-import org.spongepowered.common.data.holder.SpongeDataHolder;
+import org.spongepowered.common.bridge.data.DataCompoundHolder;
 import org.spongepowered.common.data.provider.nbt.NBTDataType;
 
-public interface NbtCompoundDataHolder extends SpongeDataHolder {
+/**
+ * Simple mutable data holder wrapper around a nbt compound. {@link org.spongepowered.common.bridge.data.CustomDataHolderBridge} is mixed in.
+ * Used for preparing data for Immutable data holders like {@link org.spongepowered.api.entity.EntitySnapshot}
+ */
+public class SimpleNbtDataHolder implements DataCompoundHolder, SpongeMutableDataHolder {
+    private CompoundNBT nbt;
+    private final NBTDataType dataType;
+
+    public SimpleNbtDataHolder(CompoundNBT nbt, NBTDataType dataType) {
+        this.nbt = nbt;
+        this.dataType = dataType;
+    }
+
+    @Override
+    public CompoundNBT data$getCompound() {
+        return this.nbt;
+    }
 
-    /**
-     * The NBT compound which contains backing
-     * data for this data holder.
-     *
-     * @return The nbt compound
-     */
-    CompoundNBT getNbtCompound();
+    @Override 
+    public void data$setCompound(CompoundNBT nbt) {
+        this.nbt = nbt;
```

Comment:
I'd personally prefer this is always copied in.

Reply:
This is basicly just a helper DataHolder for building Immutable Dataholders.
Only usage atm. SpongeEntitySnapshotBuilder
```
            final SimpleNbtDataHolder dataHolder = new SimpleNbtDataHolder(this.compound, NBTDataTypes.ENTITY);
            dataHolder.copyFrom(this.manipulator);
            this.compound = dataHolder.data$getCompound();
```

And the compound is then always copied when the builder builds the EntitySnapshot